plugins/modules/ec2_asg.py

```diff
@@ -835,8 +855,9 @@ def get_properties(autoscaling_group):
     properties['vpc_zone_identifier'] = autoscaling_group.get('VPCZoneIdentifier')
     raw_mixed_instance_object = autoscaling_group.get('MixedInstancesPolicy')
     if raw_mixed_instance_object:
-        properties['mixed_instances_policy_full'] = camel_dict_to_snake_dict(raw_mixed_instance_object)
-        properties['mixed_instances_policy'] = [x['InstanceType'] for x in raw_mixed_instance_object.get('LaunchTemplate').get('Overrides')]
+        properties['mixed_instances_policy'] = camel_dict_to_snake_dict(raw_mixed_instance_object)
```

Comment:
Unfortunately this breaks backwards compatibility as to what is returned in the `mixed_instances_policy` dict. Now it's actually returning the entire mixed_instances_policy in contrary to before where only a subsection of the mixed policy was returned, which is misleading anyhow and could sometimes be empty despite still using a mixed_instances_policy. Also the same information was already available in mixed_instances_full which would already return the entire policy.

So in my opinion it was better to fix this now before too many people rely on this 'hack'.

Reply:
hmmm. So the information was already completely returned under the `_full` key.  
Basically I'm fine with when we're dropping the key that returns less information.  

And I'm not sure if it is allowed this way, when 3.0.0 is the next release and we backport it to 2.2.0

cc  @tremble

Reply:
Just to be clear, I am not dropping any keys here, I am just returning what the name of the key mixed_instances_policy is advertising (entire policy). Also I am leaving the mixed_instances_policy_full intact to make this backwards compatible at least.

Reply:
So definitely I think what you are doing makes sense but from what I've seen on past changes, the pattern tends to be marking as deprecated first then later on making the breaking changes, so I'd maybe stick a deprecation warning on it for now, detailing that it will be changing to include the whole policy and that `mixed_instances_policy_full` will later be removed once `mixed_instances_policy` returns the full policy.

But I'll defer to the others on their views.

Reply:
The question is also what should happen during the deprecation phase. As the return value of that mixed_instances_policy would often be just empty with the current implementation, unless there is actual instance types being defined in the input. Previously this wasn't an issue as that was the only use-case the module could handle, but now with allowing a more complex policy this "bug" becomes more obvious. Hence my suggestion to fix it more like a bug, rather then going a deprecation route here.